### PR TITLE
Add ForeignFieldAdd custom gate

### PR DESCRIPTION
**Pull request by jspada**
Expose the the `ForeignFieldAdd` gate to snarky

Re https://github.com/MinaProtocol/mina/issues/12624

### CODE REVIEW

File: src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml [hunk 1]
```diff
@@ -1331,47 +1528,253 @@ end = struct
...
+           ; Some (reduce_to_v v2c15)
+           ; Some (reduce_to_v v2c16)
+           ; Some (reduce_to_v v2c17)
+           ; Some (reduce_to_v v2c18)
+           ; Some (reduce_to_v v2c19)
+          |]
+        in
+        add_row sys vars_curr RangeCheck1 [||] ;
+        add_row sys vars_next Zero [||]
+    | Plonk_constraint.T
```
Comment (querolita) [inline src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml:1648]: The problem about this approach is that you don't give support for the final Zero gate for the bound check.

Comment (querolita) [reply src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml:1648]: Let's do this as gadget as you're suggesting

**Comment by querolita** [comment]
For future reference, here's a draft PR for chainable FFAdd with final ffadd support for the bound check -> https://github.com/MinaProtocol/mina/pull/12643

**Review by querolita — APPROVED** [review]
Approving as a minimal FFAdd gate, but we should make it chainable as a gadget in the future in a similar fashion of draft https://github.com/MinaProtocol/mina/pull/12643

File: src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml (hunk 1 above)
Comment (jspada) [reply src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml:1648]: Checked with Matthew and the circuit or gadget developer can modify it from zero subsequently.

**Comment by jspada** [comment]
!ci-build-me

**Pull request merged**